sql-plugin/src/main/java/com/nvidia/spark/rapids/ColumnCastUtil.scala

```diff
@@ -43,16 +43,17 @@ object ColumnCastUtil extends Arm {
    * everything in the returned collection of AutoCloseable values.
    *
    * @param cv the view to be updated
+   * @param dt the Spark's data type of the input view (if applicable)
    * @param convert the partial function used to convert the data. If this matches and returns
    *                a updated view this function takes ownership of that view.
    * @return None if there were no changes to the view or the updated view along with anything else
    *         that needs to be closed.
    */
-  def deepTransformView(cv: ColumnView)
-      (convert: PartialFunction[ColumnView, ColumnView]):
+  def deepTransformView(cv: ColumnView, dt: Option[DataType] = None)
```

Comment:
Don't we need to update the recursive calls to deepTransformView below to include the child DataTypes if it was set? I think nested DecimalType columns are not going to be properly updated.

Reply:
Wait, `DataTypes` can also be nested? I thought not.

Reply:
So if we have a nested type `GpuColumnVector` like `struct<int, double>`, is its `dataType` also a nested struct like `StructType<IntegerType, DoubleType>`?

Reply:
Mostly.  [StructType](https://github.com/apache/spark/blob/master/sql/catalyst/src/main/scala/org/apache/spark/sql/types/StructType.scala) has a  `fileds` array that holds a StructField, which holds the name and type of each child. ListType holds the child type it wraps and MapType holds a key type and a value type.

When we create a GpuColumnVector from a cudf.ColumnVector we have asserts to verify that the types match what is expected, but those asserts are off in production. You can assume that they will match, and if they do not you will probably get a class cast exception trying to cast a DataType to the wrong child type.